### PR TITLE
fix: improve validation when uploading image to pineapple

### LINKED ISSUE #4109: Limit the upload of large files and different extensions
```[tasklist]
### Tasks
- [x] Display error if the file size is more than 1MB
- [x] Only allow file extensions `.jpg`, `.jpeg`, `.png`, and `.gif`
- [ ] ~~[Optional] Show error if user tries to upload image with width/height greater than `1000px`~~
- [x] [Optional] Display human-readable error in case an error is returned from pineapple (may depend on https://github.com/snapshot-labs/pineapple/issues/140)
```

### PR DESCRIPTION
**Pull request by wa0x6e**
### Issues

Current image upload through pineapple does not validate file size.
Pineapple is returning an error for file > 1MB

We should enforce the file size check on client side to avoid wasting bandwidth, and saving pineapple resources.

Also, there's no error message when user avatar upload fail when updating

Fixes #4109 

### Changes 

1. Add client-side file size validation when uploading image.
2. Error when uploading the user avatar is now displayed in a notification
3. When pineapple.js returns an human readable error, show it instead of the generic "Something went wrong"

### How to test

1. Go to your profile
2. Upload an avatar > 1MB
3. It should display an error, without starting the upload 
4. Rename a non-image file with the .png extension, and upload it
5. It should display a "unsupported file type" error, which is coming directly from pineapple

### Self-review checklist
- [x] I have performed a full self-review of my changes
- [x] I have tested my changes on a preview deployment
- [ ] I have tested my changes on different screen sizes (sm, md)
- [ ] I have tested my changes on a custom domain

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

### CODE REVIEW

File: src/composables/useImageUpload.ts [hunk 1]
```diff
@@ -27,17 +27,28 @@ export function useImageUpload() {
...
+      return;
+    }
     formData.append('file', file);
     try {
       const receipt = await pin(formData);
+
+      if (receipt.error) {
+        imageUploadError.value = receipt.error.message;
+        isUploadingImage.value = false;
+        return;
```
Comment (samuveth) [inline src/composables/useImageUpload.ts:49]: Should we not throw an error instead of returning?

Comment (wa0x6e) [reply src/composables/useImageUpload.ts:49]: I am keeping the same flow as the error from the file type

**Review by samuveth — APPROVED** [review]
tACK

**Comment by samuveth** [comment]
@wa0x6e can you resolve conflicts and merge

**Review by samuveth — APPROVED** [review]
tACK

**Comment by samuveth** [comment]
@wa0x6e can you fix conflicts and merge

**Comment by wa0x6e** [comment]
> @wa0x6e can you fix conflicts and merge
> 
> edit: Oh, I wrote that already :D

Done

**Pull request merged**